### PR TITLE
Use crates index instead of crates.io api

### PR DESCRIPTION
**Pull request by Shnatsel**
Well, that was easy.

### CODE REVIEW

**Review by tarcieri — APPROVED** [review]

**Comment by Shnatsel** [comment]
After this change the entire advisory db is linted in 130ms with a warm cache, 3.7s with a cold cache on an HDD.

Not sure how impactful fetching crates.io index will be on CI execution time, however.

**Comment by lnicola** [comment]
In rust-analyzer we cache the cargo index IIRC.

File: admin/src/linter.rs [hunk 1]
```diff
@@ -34,15 +34,15 @@ impl Linter {
     /// Create a new linter for the database at the given path
     pub fn new(repo_path: impl Into<PathBuf>) -> Result<Self, Error> {
         let repo_path = repo_path.into();
-        let http_client = ureq::AgentBuilder::new()
-            .user_agent("RustSec advisory database linter")
-            .timeout(Duration::from_secs(20))
-            .build();
+        let crates_index = crates_index::Index::new_cargo_default();
+        if !crates_index.exists() {
+            crates_index.retrieve()?;
```
Comment (lnicola) [inline admin/src/linter.rs:39]: Is there a way to update it in case it exists, but it's stale?

Comment (Shnatsel) [reply admin/src/linter.rs]: Good catch! Turns out the crate examples are misleading. Fixed!

**Review by lnicola — APPROVED** [review]

**Review by alex — APPROVED** [review]

**Pull request merged**